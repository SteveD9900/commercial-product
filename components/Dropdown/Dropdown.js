import React, { useState } from "react";
import styles from "./Dropdown.module.scss";

export default function Dropdown(props) {
    const [selectedType, setSelectedType] = useState("All");
    const options = [
      "All",
      "Beer",
      "Wine",
      "Spirits",
      "Cider"
    ];

    const onChangeHandler = (e) => {
      setSelectedType(e.target.value);
      props.filterItems(e.target.value);
    }

    return (
        <div className={styles.Dropdown}>
            <span>Filter By</span>
            <select
              className={styles.Dropdown}
              value={selectedType}
              onChange={onChangeHandler}
            >
              {options.map((option) => (
                <option key={option} name={option} value={option}>{option}</option>
              ))}

            </select>
        </div>
    );
  }