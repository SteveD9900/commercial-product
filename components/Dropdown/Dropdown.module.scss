.Dropdown{
    display: inline-block;
}
.SelectorStyle {
    width: 80px;
}