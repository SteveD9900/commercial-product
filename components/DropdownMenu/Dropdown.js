import React, { useState } from "react";
import styles from "./Dropdown.module.scss";

export default function Dropdown(props) {
    const [selectedType, setSelectedType] = useState("All");
    const options = [
        "All",
        "Beer",
        "Wine",
        "Spirits",
        "Cider"
      ];
    return (
        <div className={styles.Dropdown}>
            <span>Filter By</span>
            <select
              className={styles.Dropdown}
              value={selectedType}
              onChange={(e) => {
                setSelectedType(e.target.value);
              }}
            >
              {options.map((option) => (
                <option key={option} name={option} value={option}>{option}</option>
              ))}

            </select>
        </div>
    );
  }