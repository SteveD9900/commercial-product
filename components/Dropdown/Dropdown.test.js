import Dropdown from './Dropdown';
import { shallow } from 'enzyme'

let wrapper;
let typeList = ["All"];

beforeEach(() => {
    wrapper = shallow(<Dropdown options={typeList}/>);
});

describe('<Dropdown /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});