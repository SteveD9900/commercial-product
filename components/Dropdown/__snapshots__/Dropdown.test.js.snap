// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> rendering renders correctly 1`] = `ShallowWrapper {}`;
